flogger/flogger-system-backend/src/test/kotlin/com/google/common/flogger/backend/system/SimpleLogRecordSpec.kt

```diff
@@ -0,0 +1,281 @@
+        private val ID_KEY = MetadataKey.single("id", String::class.java)
+        private val PATH_KEY = object : MetadataKey<String>("path", String::class.java, true) {
+            override fun emitRepeated(values: Iterator<String>, out: KeyValueHandler) {
+                val joined = values.asSequence().joinToString("/")
+                out.handle(label, joined)
+            }
+        }
+        private const val LITERAL = "literal message"
+    }
+
+    @Nested inner class
+    `return the provided` {
+
+        @Test
+        fun `log level`() {
+            val level = Level.FINER
+            val data = FakeLogData.of("") .setLevel(level)
+            val record = SimpleLogRecord.create(data, Metadata.empty())
+            record.level shouldBe level
+        }
+
+        @Test
+        fun `literal message`() {
+            val data = FakeLogData.of(LITERAL)
+            val record = SimpleLogRecord.create(data, Metadata.empty())
+            record.message shouldBe LITERAL
+            record.parameters.shouldBeEmpty()
+        }
+
+        @Test
+        fun `logger name and log site info`() {
+            val data = FakeLogData.of("")
+            val record = SimpleLogRecord.create(data, Metadata.empty())
+            record.loggerName shouldBe data.loggerName
+            record.sourceClassName shouldBe data.logSite.className
+            record.sourceMethodName shouldBe data.logSite.methodName
+        }
+
+        @Test
+        fun `'Throwable' cause`() {
+            val cause = Throwable("Goodbye World")
+            val data = FakeLogData.of("").addMetadata(Key.LOG_CAUSE, cause)
+            val record = SimpleLogRecord.create(data, Metadata.empty())
+            record.thrown shouldBeSameInstanceAs cause
+        }
+    }
+
+    @Nested inner class
+    `append to message` {
+
+        @Test
+        fun `log statement metadata`() {
+            val timestampNanos = 123456789000L
+            val data = FakeLogData.of(LITERAL)
+                .setTimestampNanos(timestampNanos)
+                .addMetadata(COUNT_KEY, 23)
+                .addMetadata(ID_KEY, "test ID")
+            val record = SimpleLogRecord.create(data, Metadata.empty())
+            record.message shouldBe "$LITERAL [CONTEXT count=23 id=\"test ID\" ]"
+            record.instant shouldBe ofEpochMilli(NANOSECONDS.toMillis(timestampNanos))
+            record.parameters.shouldBeEmpty()
+        }
+
+        @Test
+        fun `merged scope and log statement metadata`() {
+            val scope = FakeMetadata()
+                .add(PATH_KEY, "foo")
+                .add(COUNT_KEY, 23)
+                .add(PATH_KEY, "bar")
+            val data = FakeLogData.of(LITERAL)
+                .addMetadata(ID_KEY, "foo")
+                .addMetadata(PATH_KEY, "baz")
+            val record = SimpleLogRecord.create(data, scope)
+            record.message shouldBe "$LITERAL [CONTEXT path=\"foo/bar/baz\" count=23 id=\"foo\" ]"
```

Comment:
This is a common issue in these tests, but let me write a single comment here, since it's, again, illustrative.

When composing expected values we should operate with those original variables set as inputs for the code-under-test. Otherwise, the tests are fragile. And expensive to support.

Please adjust here and in other similar places.